controllers/policyserver_controller_test.go

```diff
@@ -53,15 +53,17 @@ var _ = Describe("PolicyServer controller", func() {
 			}, timeout, pollInterval).ShouldNot(
 				HaveField("Finalizers", ContainElement(constants.KubewardenFinalizer)),
 			)
+		})
 
+		AfterEach(func() {
 			// It's necessary remove the test finalizer to make the
 			// BeforeEach work as extected. Otherwise, the policy service
 			// creation will not work as expected
 			policyServer, err := getTestPolicyServer(policyServerName)
 			Expect(err).Should(Succeed())
 			controllerutil.RemoveFinalizer(policyServer, IntegrationTestsFinalizer)
 			err = reconciler.Client.Update(ctx, policyServer)
-			Expect(err).ToNot(HaveOccurred())
+			Expect(err).ShouldNot(HaveOccurred())
```

Comment:
This is a nitpick, in this case using `ToNot` leads to a correct English sentence: "expect err to not have occurred". While `ShouldNot` leads to something that isn't

Reply:
Also, given this is now an after each block, we don't know what is actually done inside of the test -> we can't assume the deletion of the PolicyServer was requested.

The code should be something like:

```
err := deletePolicyServer()
expect(err).ToNot(HaveOccurred())

removeFinalizer(policyServer, IntegrationTestFinalizer)
err = update(policyServer)
expect(err).ToNot(HaveOccurred())

eventually(wait for PolicyServer to not be found)
```

Reply:
I don't think we need to do all of that. Notice that I've added the `AfterEach` on the `with no assigned policies` context. Not in the most external. If I've added it in the same level of the current `BeforeEach` other tests will failed. Because the policy server will be recreated on every `It` call (test/spec). But we do not want that. Because the tests under the `it has assigned policies` context need the state of the previous tests (`It` calls). 

However, I think that a eventually call is nice thing to add there.

Reply:
I see, I take my comment back